Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_browser_driver.py

```diff
@@ -41,6 +42,42 @@ def restore_env_after_all_testing(self):
     def close_browsers(self):
         self._terminate_processes(self.process_name, self.bundle_id)
 
+    @contextmanager
+    def profile(self, output_path, profile_filename, profiling_interval, trace_type='profile', timeout=300):
+        trace_process = None
+        additional_trace_arguments = {
+            'full': ['--type=full'],
+            'profile': ['--type=profile'],
+        }
+        if trace_type not in additional_trace_arguments:
+            _log.error('`{}` is not a valid type of trace. Defaulting to `profile`.'.format(trace_type))
+            trace_type = 'profile'
+        try:
+            _log.info('Gathering traces')
```

Comment:
Worth add more logging to tell user to type host machine admin password if get a prompt.